### DISCUSSION TITLE
Lost Grid Status & PW capacity , Voltages with 22.18.3 21c0ad81 Update

### DISCUSSION THREAD

**Discussion by s0schen** [jasonacox/Powerwall-Dashboard#47] [Q&A] [answered]
Anyone else with this problem?  

I also noticed this new statement in Customer Facing Gateway Summary Page: 
SOLAR ASSEMBLY
Stop system first to see part number and serial of Powerwall+ solar assembly.

I also see some log Errors:

> pyPowerwall [0.4.0] Proxy Server [t15] Started - HTTP Port 8675
> ----------------------------------------
> Exception happened during processing of request from ('172.19.0.5', 42354)
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.8/socketserver.py", line 683, in process_request_thread
>     self.finish_request(request, client_address)
>   File "/usr/local/lib/python3.8/socketserver.py", line 360, in finish_request
>     self.RequestHandlerClass(request, client_address, self)
>   File "/usr/local/lib/python3.8/socketserver.py", line 747, in __init__
>     self.handle()
>   File "/usr/local/lib/python3.8/http/server.py", line 427, in handle
>     self.handle_one_request()
>   File "/usr/local/lib/python3.8/http/server.py", line 415, in handle_one_request
>     method()
>   File "server.py", line 182, in do_GET
>     fcv["PW%d_PINV_Fout" % idx] = d['PINV_Fout']
> KeyError: 'PINV_Fout'
> ----------------------------------------
> ----------------------------------------
> Exception happened during processing of request from ('172.19.0.5', 42344)
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.8/socketserver.py", line 683, in process_request_thread
>     self.finish_request(request, client_address)
>   File "/usr/local/lib/python3.8/socketserver.py", line 360, in finish_request
>     self.RequestHandlerClass(request, client_address, self)
>   File "/usr/local/lib/python3.8/socketserver.py", line 747, in __init__
>     self.handle()
>   File "/usr/local/lib/python3.8/http/server.py", line 427, in handle
>     self.handle_one_request()
>   File "/usr/local/lib/python3.8/http/server.py", line 415, in handle_one_request
>     method()
>   File "server.py", line 202, in do_GET
>     pod["PW%d_POD_ActiveHeating" % idx] = int(d['POD_ActiveHeating'])
> KeyError: 'POD_ActiveHeating'

**Comment by jasonacox** [comment]
Hi @s0schen - My system upgraded last night as well.  Originally all data stopped and upon investigation, I saw "API rate limit" errors on the Powerwall gateway portal. Somehow during the upgrade, my latest password reset reverted and I had to update it again and things restored.  During my troubleshooting I reset the gateway (not sure if that matters).  Right now, I'm not seeing any missing data, but I do see the note you mention on the Summary Page.

Try this to see if you can get that data:

```bash
# change localhost to the machine running the pypowerwall
http://localhost:8675/vitals
http://localhost:8675/soe
http://localhost:8675/pod
```

Also, you can try to access it directly via pypowerwall in python:

```python
import pypowerwall
pw = pypowerwall.Powerwall(host,password,email)
pw.grid()
pw.level()
pw.vitals()
```

**Reply by jasonacox** [reply]
You might try rebooting the backup switch.  I did this during my troubleshooting.  See page 15 of the [manual here](https://tesla-cdn.thron.com/static/FXC5ME_Tesla_Powerwall__Owner_Manual_EN-NA__1_2NTCVX.pdf?xseo=&response-content-disposition=inline%3Bfilename%3D%22powerwall-plus-owners-manual-en-na.pdf%22).

**Reply by s0schen** [reply]
vitals and soe both worked.

http://localhost:8675/pod  returns Err Empty Response

No Backup Switch here, as they installed the Gateway.  I'll try rebooting it. Thanks

**Comment by s0schen** [comment, marked as answer]
Problem Solved.   Used Installer login to look at the system and discovered that one of four powerwalls was in a failed update mode and was disabled.  

Luckily problem was fixed by running install wizard and it forced the powerwall update which was successful this time and brought the fourth powerwall back online.   Unfortunately I've had this problem in the past.

**Reply by jasonacox** [reply]
Awesome!  Thanks for the update @s0schen - could be helpful for anyone else seeing this behavior.  Out of curiosity, did you happen to see any "Alerts" in the vitals payload (or via `http://localhost:8675/alerts` which I forgot to mention) that would indicate that PW update was in a bad state?  I just wonder how the gateway broadcasts that error state.

**Reply by s0schen** [reply]
Everything looks ok now.  I'll have to check next time.  

There are these two lines:
"PodCommissionTime",
"PodCommissionTimeError",

**Reply by jasonacox** [reply]
Thanks @s0schen !  I added PodCommissionTimeError to the list of errors I'm tracking here: https://github.com/jasonacox/pypowerwall#devices-and-alerts in case it helps someone else troubleshoot in the future.

**Reply by cstout1** [reply]
I'm going through a similar failed upgrade experience here and have one additional undocumented alert:
BatteryDisabled

**Reply by jasonacox** [reply]
@cstout1 What firmware version do you have (did you have)?

**Reply by cstout1** [reply]
Firmware: 23.12.2
